### PR TITLE
Get kine from official binaries & follow the upstream golang version

### LINKED ISSUE #482: Get kine from official binaries, no need to build it from source
**What happened?**
Actually, it's not a bug. :smiley: 
Just to get the kine from official binaries (https://github.com/k3s-io/kine/releases), no need to build it from source.

**How To Reproduce**
Check the makefile:
https://github.com/k0sproject/k0s/blob/b25e26d3f4d86fb667fd52ff9f6955ca381789a5/embedded-bins/Makefile#L91-L98

**Expected behavior**
Just download the kine, then save compilation and testing time.

**Screenshots & Logs**
None.

**Additional context**
None.

/area packaging

### PR DESCRIPTION
**Pull request by warmchang**
Signed-off-by: William Zhang <warmchang@outlook.com>

---
name: Pull Request
about: Create a Pull Request
title: ''
labels: ''
assignees: ''

---

**Issue**
Fixes #482

**What this PR Includes**
1. Get kine from official binaries, no need to build it from source.
2. Follow the upstream golang version (https://github.com/k3s-io/kine/blob/v0.6.0/Dockerfile#L1).

### CODE REVIEW

**Comment by ncopa** [comment]
I didn't notice that kine started to ship pre-compiled binaries.

Do you think you do the version changes and Dockerfile fixes (including golang update) in separate commit(s)? Maybe even in separate PR. And maybe also reabase (due to https://github.com/k0sproject/k0s/pull/480)

It looks good otherwise.

Thanks!

File: embedded-bins/Makefile [hunk 1]
```diff
@@ -2,7 +2,7 @@
 runc_version = 1.0.0-rc92
 containerd_version = 1.4.3
 kubernetes_version = 1.19.4
-kine_version = 0.5.1
+kine_version = 0.6.0
```
Comment (ncopa) [inline embedded-bins/Makefile:5]: This needs to be rebase due to https://github.com/k0sproject/k0s/pull/480

File: embedded-bins/kine/Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.13-alpine AS build
+FROM golang:1.15-alpine AS build
```
Comment (ncopa) [inline embedded-bins/kine/Dockerfile:1]: I'd like this in a separate commit

File: Dockerfile [hunk 3]
```diff
@@ -2,13 +2,13 @@ FROM alpine:3.12
...
 
-RUN curl -LO https://storage.googleapis.com/kubernetes-release/release/v1.19.0/bin/linux/amd64/kubectl \
+RUN curl -LO https://storage.googleapis.com/kubernetes-release/release/v1.19.4/bin/linux/amd64/kubectl \
        && chmod +x ./kubectl \
        && mv ./kubectl /usr/local/bin/kubectl
 ENV KUBECONFIG=/var/lib/k0s/pki/admin.conf
 
 ADD docker-entrypoint.sh /entrypoint.sh
-ADD ./k0s /usr/local/bin/k0s 
+ADD ./k0s /usr/local/bin/k0s
```
Comment (ncopa) [inline Dockerfile:11]: I'd like those two changes in separate commit.

**Review by ncopa — CHANGES_REQUESTED** [review]

File: embedded-bins/kine/Dockerfile (hunk 2 above)
Comment (warmchang) [reply embedded-bins/kine/Dockerfile:1]: Done

File: embedded-bins/Makefile (hunk 1 above)
Comment (warmchang) [reply embedded-bins/Makefile:5]: Done

**Review by ncopa — APPROVED** [review]

**Comment by ncopa** [comment]
Not sure how you have set up remotes, but if your `origin` is `https://github.com/k0sproject/k0s`
I wonder if you could you do something like:
```
git fetch origin
git rebase origin/main
git push -f       # push to your fork
```

**Comment by warmchang** [comment]
@ncopa I rebased the wrong upstrem, that's why the commits are confused before, but now it's ok (by cherry-pick), pls check it. 😃

**Review by ncopa — APPROVED** [review]

**Review by ncopa — APPROVED** [review]

**Pull request merged**